src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ConstructorWriter.java

```diff
@@ -189,13 +189,15 @@ protected Content getConstructorHeaderContent(ExecutableElement constructor) {
         Content content = new ContentBuilder();
         var heading = HtmlTree.HEADING(Headings.TypeDeclaration.MEMBER_HEADING,
                 Text.of(name(constructor)));
+        HtmlId memberAnchor = htmlIds.forMember(constructor);
         HtmlId erasureAnchor = htmlIds.forErasure(constructor);
-        if (erasureAnchor != null) {
+        if (erasureAnchor != null
+                && !erasureAnchor.name().equals(memberAnchor.name())) {
             heading.setId(erasureAnchor);
         }
         content.add(heading);
         return HtmlTree.SECTION(HtmlStyle.detail, content)
-                .setId(htmlIds.forMember(constructor));
+                .setId(memberAnchor);
```

Comment:
It's a bit disappointing that more of this isn't in `HtmlIds`.
It feels like it perpetuates the original ugly code.

Would it make sense for `htmlIds` to return a record/pair containing both the `memberId` and `erasureId` for `ExecutableElement` ?

Reply:
Please have a look at the recent update to see if it's to your liking. Thanks.